### ISSUE TITLE
CIOULoss is Nan

### ISSUE THREAD

**Issue by qixitan** [Megvii-BaseDetection/YOLOX#1565]
After using ciouloss in yolox, ciouloss is nan，ciouloss code is as follows

elif self.loss_type == 'ciou':
    c_tl = torch.min((pred[:, :2] - pred[:, 2:] / 2), (target[:, :2] - target[:, 2:] / 2))
    c_br = torch.max((pred[:, :2] + pred[:, 2:] / 2), (target[:, :2] + target[:, 2:] / 2))
    w_c = (c_br - c_tl)[:, 0]
    h_c = (c_br - c_tl)[:, 1]
    c = w_c ** 2 + h_c ** 2
    w_d = (pred[:, :2] - target[:, :2])[:, 0]
    h_d = (pred[:, :2] - target[:, :2])[:, 1]
    d = w_d ** 2 + h_d ** 2
    diou = iou - d / c.clamp(1e-16)
    w_gt = target[:, 2]
    h_gt = target[:, 3]
    w = pred[:, 2]
    h = pred[:, 3]
    with torch.no_grad():
        arctan = torch.atan(w_gt / h_gt.clamp(1e-16)) - torch.atan(w / h.clamp(1e-16))
        v = (4 / (math.pi ** 2)) * torch.pow(arctan, 2)
        s = 1 - iou
        alpha = v / (s + v).clamp(1e-16)
    ciou = diou - alpha * v
    loss = 1 - ciou.clamp(min=-1.0, max=1.0)

**Comment by Monkey-D-Luffy-star** [comment]
请问你解决这个问题了吗？

**Comment by qixitan** [comment]
可以尝试关掉fp16,
